fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccrualWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,253 @@
+                    "Failed to process accrual entries for savings account: " + savingsAccountId, errors);
+        }
+        return CommandProcessingResult.empty();
+    }
+
+    private void addAccrualTransactions(SavingsAccount savingsAccount, final LocalDate fromDate, final LocalDate tillDate,
+            final Integer financialYearBeginningMonth, final boolean isSavingsInterestPostingAtCurrentPeriodEnd, final MathContext mc) {
+        final Set<Long> existingTransactionIds = new HashSet<>();
+        final Set<Long> existingReversedTransactionIds = new HashSet<>();
+        Boolean isNegativeBalance = false;
+        existingTransactionIds.addAll(savingsAccount.findExistingTransactionIds());
+        existingReversedTransactionIds.addAll(savingsAccount.findExistingReversedTransactionIds());
+
+        List<LocalDate> postedAsOnTransactionDates = savingsAccount.getManualPostingDates();
+        final SavingsPostingInterestPeriodType postingPeriodType = SavingsPostingInterestPeriodType
+                .fromInt(savingsAccount.getInterestCalculationType());
+
+        final SavingsCompoundingInterestPeriodType compoundingPeriodType = SavingsCompoundingInterestPeriodType
+                .fromInt(savingsAccount.getInterestPostingPeriodType());
+
+        final SavingsInterestCalculationDaysInYearType daysInYearType = SavingsInterestCalculationDaysInYearType
+                .fromInt(savingsAccount.getInterestCalculationDaysInYearType());
+
+        final List<LocalDateInterval> postingPeriodIntervals = this.savingsHelper.determineInterestPostingPeriods(fromDate, tillDate,
+                postingPeriodType, financialYearBeginningMonth, postedAsOnTransactionDates);
+
+        final List<PostingPeriod> allPostingPeriods = new ArrayList<>();
+        final MonetaryCurrency currency = savingsAccount.getCurrency();
+        Money periodStartingBalance = Money.zero(currency);
+
+        final SavingsInterestCalculationType interestCalculationType = SavingsInterestCalculationType
+                .fromInt(savingsAccount.getInterestCalculationType());
+        final BigDecimal interestRateAsFraction = savingsAccount.getEffectiveInterestRateAsFraction(mc, tillDate);
+        final Collection<Long> interestPostTransactions = this.savingsHelper.fetchPostInterestTransactionIds(savingsAccount.getId());
+        boolean isInterestTransfer = false;
+        final Money minBalanceForInterestCalculation = Money.of(currency, savingsAccount.getMinBalanceForInterestCalculation());
+        List<SavingsAccountTransactionDetailsForPostingPeriod> savingsAccountTransactionDetailsForPostingPeriodList = savingsAccount
+                .toSavingsAccountTransactionDetailsForPostingPeriodList();
+        for (final LocalDateInterval periodInterval : postingPeriodIntervals) {
+            if (DateUtils.isDateInTheFuture(periodInterval.endDate())) {
+                continue;
```

Comment:
Do we really need to check all the rest of the intervals after finding the first which is in the future?